lib/mayaUsd/fileio/orphanedNodesManager.cpp

```diff
@@ -259,6 +259,18 @@ bool OrphanedNodesManager::empty() const { return pulledPrims().root()->empty();
 
 void OrphanedNodesManager::restore(Memento&& previous) { _pulledPrims = previous.release(); }
 
+bool OrphanedNodesManager::isOrphaned(const Ufe::Path& pulledPath) const
+{
+    auto trieNode = pulledPrims().node(pulledPath);
+    if (!trieNode) {
+        // If the argument path has not been pulled, it can't be orphaned.
+        return false;
+    }
+    TF_VERIFY(trieNode->hasData());
+    // If the pull parent is visible, the pulled path is not orphaned.
+    return !getVisibilityPlug(trieNode);
```

Comment:
The only caveat I see with this approach, which we may not want to support, is: what if the user has set the node invisible by hand for whatever reason? Here we take for granted that invisible means orphaned.

Reply:
If you touch the pull parent, which by default is not visible in the Outliner, you're voiding your warranty...  So yes, invisible pull parent means orphaned.